librz/main/rizin.c

```diff
@@ -389,6 +391,7 @@ RZ_API int rz_main_rizin(int argc, const char **argv) {
 	rz_signal_sigmask(SIG_BLOCK, &sigBlockMask, NULL);
 #endif
 
+	rz_demangler_plugin_init();
```

Comment:
Why is this needed? I think it should not be here... maybe somewhere inside RzCore? Please look at how all the other plugins are initialized and do the same, this feels wrong.

Reply:
no, i disagree.
it should be initialized by the plugin only once and i want to avoid to implement ref counting for this.
i do not want to place this into RzCore because rz-bin needs it since it provides demangling and you can create n-th instances of the RzCore.
Also it is required to have dynamically allocated lists where plugins of this type are registered when loading shared libs.

Reply:
I'm not convinced. Look at how other plugins do their initialization. Look at config.h.in and maybe let's add a `RZ_DEMANGLER_STATIC_PLUGINS` there with the list of plugins you need to build. You can easily define them in meson.build instead of at runtime.

Reply:
it does not change the fact that they cannot be initialized by RzCore

Reply:
You do not need to initialize/finalize anything if you just move the list in RZ_DEMANGLER_STATIC_PLUGINS, like what's done for all the other plugins types. It is no different than a RzBin plugin, needed by rz-bin.

Reply:
no. all the plugins are still initialized when you call the rz_<something>_new();